### DISCUSSION TITLE
Nginx crash with Upstream Timed out error

### DISCUSSION THREAD

**Discussion by scaza11** [leokhoa/laragon#487] [Q&A]
Hello All,

I use laragon on Windows server 2008 R2 to run GLPI

Regularly Nginx (version 1.14.0) crash and i have to reload nginx
The file error.log contains this line:
[error] 4596#2480: *8324 upstream timed out (10060: FormatMessage() error:(15105)) while connecting to upstream

the file auto.glpi.conf contains this:

server {
    listen 80;
    listen 443 ssl;
    server_name glpi *.glpi;
    root "C:/laragon/www/glpi";
    
    index index.html index.htm index.php;
 
    location / {
        try_files $uri $uri/ /index.php$is_args$args;
		autoindex on;
    }
    
    location ~ \.php$ {
        include snippets/fastcgi-php.conf;
        fastcgi_pass php_upstream;		
        #fastcgi_pass unix:/run/php/php7.0-fpm.sock;
    }

    # Enable SSL
    ssl_certificate "C:/laragon/etc/ssl/laragon.crt";
    ssl_certificate_key "C:/laragon/etc/ssl/laragon.key";
    ssl_session_timeout 5m;
    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
    ssl_ciphers ALL:!ADH:!EXPORT56:RC4+RSA:+HIGH:+MEDIUM:+LOW:+SSLv3:+EXP;
    ssl_prefer_server_ciphers on;
	
	
    charset utf-8;
	
    location = /favicon.ico { access_log off; log_not_found off; }
    location = /robots.txt  { access_log off; log_not_found off; }
    location ~ /\.ht {
        deny all;
    }
}

# This file is auto-generated.
# If you want Laragon to respect your changes, just remove the [auto.] prefix

Do you know where the problem could come from?

**Comment by samuelpatro** [comment]
Same issue here it does randomly on ajax requests.

**Comment by amadeann** [comment]
I had a similar issue, and fixed it by changing `Upstream=2` to `Upstream=5` in `C:\laragon\usr\laragon.ini` file.

A comment in this file:
`C:\laragon\etc\nginx\php_upstream.conf`
mentioned this:
> `# if you want to increase the number of PHP processes, put this line (without #) on usr/laragon.ini, under [nginx]`

**Reply by tharlab** [reply]
nice, got same problem and ill try this

**Reply by ArdaGnsrn** [reply]
Thank you for your answer, but I'm fine with that error. Just when you've completely forgotten, it suddenly appears and reminds you of itself. 🐳🐳

**Reply by fatihozpolat** [reply]
> Thank you for your answer, but I'm fine with that error. Just when you've completely forgotten, it suddenly appears and reminds you of itself. 🐳🐳

Use apache. There is no problem. 🐳

**Reply by 0x1881** [reply]
Don't use Apache. It creates critical problems locally. 🐳

**Reply by cemtekinn** [reply]
Apache is better than Nginx 🐳🐳🐳🐳 Use Apache, let it be used 😉😉😉😉😉

**Reply by ArdaGnsrn** [reply]
> Don't use Apache. It creates critical problems locally. 🐳

I totally agree. Apache is disappointing. Nginx, on the other hand, succeeds in making you happy with its little surprises. 🐳🐳

**Reply by ArdaGnsrn** [reply]
> Apache is better than Nginx 🐳🐳🐳🐳 Use Apache, let it be used 😉😉😉😉😉

![image](https://github.com/leokhoa/laragon/assets/52893305/ae940fd4-f266-4c9a-a458-3bfb6d4bffc2)
